### PR TITLE
fix for border render

### PR DESCRIPTION
**Pull request by ssaket**
Fix for forum link - http://www.html5gamedevs.com/topic/41694-general-gui-troubles/?tab=comments#comment-237278

### CODE REVIEW

File: gui/src/2D/controls/inputText.ts [hunk 1]
```diff
@@ -923,19 +923,17 @@ export class InputText extends Control implements IFocusableControl {
...
-            if (this._thickness) {
-                if (this.color) {
-                    context.strokeStyle = this.color;
-                }
-                context.lineWidth = this._thickness;
-
-                context.strokeRect(this._currentMeasure.left + this._thickness / 2, this._currentMeasure.top + this._thickness / 2,
-                    this._currentMeasure.width - this._thickness, this._currentMeasure.height - this._thickness);
+        }
+        context.restore();
```
Comment (deltakosh) [inline gui/src/2D/controls/inputText.ts]: This is probably a bug. The restore should be inside the if focused as the associated save is also inside a if focused.

You can test it with the first pg from the initial thread on the forum

**Comment by ssaket** [comment]
Please don't merge, working on adding a new property onFocusBorderColor, which by default should be "blue".

File: gui/src/2D/controls/inputText.ts (hunk 1 above)
Comment (ssaket) [reply gui/src/2D/controls/inputText.ts]: if I put it inside it creates incomplete border

Comment (ssaket) [reply gui/src/2D/controls/inputText.ts]: screenshot 2018-12-07 at 8 10 52 pm
for PG - https://www.babylonjs-playground.com/#YP180Z
@deltakosh - I don't know the cause yet.

**Comment by ssaket** [comment]
@deltakosh , Sorry not sure about the context handling, Therefore, closing this pull request.

**Comment by deltakosh** [comment]
Your fix was good. Just let the restore inside the if :)

**Pull request closed without merging**